src/components/buttons/ButtonSolid.tsx

```diff
@@ -211,29 +221,35 @@ export const ButtonSolid = React.memo(
       isPressed.value = 0;
     }, [isPressed]);
 
+    const handleOnPress = (event: GestureResponderEvent) => {
+      ReactNativeHapticFeedback.trigger("impactLight");
+      onPress(event);
+    };
+
     // Label & Icons colors
     const foregroundColor: IOColors = disabled
       ? colorMap[color]?.label?.disabled
       : colorMap[color]?.label?.default;
 
     return (
       <Pressable
+        testID={testID}
+        accessible={true}
         accessibilityLabel={accessibilityLabel}
         accessibilityHint={accessibilityHint}
+        accessibilityState={{ busy: loading }}
         accessibilityRole={"button"}
-        testID={testID}
-        onPress={onPress}
+        onPress={handleOnPress}
         onPressIn={onPressIn}
         onPressOut={onPressOut}
-        accessible={true}
         disabled={disabled}
```

Comment:
maybe we should force a disabled state while the loading is running, in this way we may prevent from any other Tap by the user, or at least if we don't need to alter the UI of the button maybe we can _disable_ the onPress event in some way.

What do you think?

Reply:
Tried to add it, but the addition disabled the scale effect applied to the Button (even when it's not in loading state)

Reply:
@CrisTofani Added a new line to avoid calling `onPress`

Reply:
What about instead changing the handleOnPress with a useCallback? 

Something like:

```typescript
    const handleOnPress = useCallback((event: GestureResponderEvent) => {
      if(isLoading) {
        return;
      }
      ReactNativeHapticFeedback.trigger("impactLight");
      onPress(event);
    }, [isLoading, onPress]);
```

Reply:
Seems cleaner. Addressed in the following commit https://github.com/pagopa/io-app-design-system/pull/99/commits/09bcb189f1364302b5a6ab9e022b0691dcd90f54